### PR TITLE
feat:Add online and ready functions based on Spring Boot Actuator

### PR DESCRIPTION
**Pull request by JinQian1024**
## What is the purpose of the change
It implements the online (to be improved) and ready functions based on Spring Boot Actuator and equivalent to the QoS functions

## Checklist
- [x] Make sure there is a [GitHub_issue](https://github.com/apache/dubbo/issues) field for the change (usually before you start working on it). Trivial changes like typos do not require a GitHub issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
- [ ] Each commit in the pull request should have a meaningful subject line and body.
- [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [ ] Check if is necessary to patch to Dubbo 3 if you are work on Dubbo 2.7
- [ ] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add sample in [dubbo samples](https://github.com/apache/dubbo-samples) project.
- [ ] Add some description to [dubbo-website](https://github.com/apache/dubbo-website) project if you are requesting to add a feature.
- [ ] GitHub Actions works fine on your own branch.
- [ ] If this contribution is large, please follow the [Software Donation Guide](https://github.com/apache/dubbo/wiki/Software-donation-guide).

### CODE REVIEW

File: dubbo-spring-boot/dubbo-spring-boot-actuator/src/main/java/org/apache/dubbo/spring/boot/actuate/autoconfigure/DubboEndpointAnnotationAutoConfiguration.java [hunk 1]
```diff
@@ -45,7 +47,7 @@
 @Configuration
 @PropertySource(
         name = "Dubbo Endpoints Default Properties",
-        value = "classpath:/META-INF/dubbo-endpoints-default.properties")
+        value = "classpath:/META-INF/dubbo-endpoints-default" + ".properties")
```
Comment (AlbumenJ) [inline dubbo-spring-boot/dubbo-spring-boot-actuator/src/main/java/org/apache/dubbo/spring/boot/actuate/autoconfigure/DubboEndpointAnnotationAutoConfiguration.java]: Why change this

File: dubbo-spring-boot/dubbo-spring-boot-actuator/src/main/java/org/apache/dubbo/spring/boot/actuate/endpoint/DubboOnlineEndpoint.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.actuate.endpoint.annotation.Endpoint;
+import org.springframework.boot.actuate.endpoint.annotation.WriteOperation;
+
+/**
+ * Dubbo Online
+ *
+ * @since 3.3.0
+ */
+@Endpoint(id = "dubboonline")
```
Comment (AlbumenJ) [inline dubbo-spring-boot/dubbo-spring-boot-actuator/src/main/java/org/apache/dubbo/spring/boot/actuate/endpoint/DubboOnlineEndpoint.java:33]: offline is also useful

**Review by AlbumenJ — CHANGES_REQUESTED** [review]

File: dubbo-spring-boot/dubbo-spring-boot-compatible/actuator/src/main/java/org/apache/dubbo/spring/boot/actuate/endpoint/metadata/DubboGetEnabledRouterMetadata.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+/**
+ * Dubbo Get Enabled Router Snapshot
+ *
+ * @since 3.3.0
+ */
+@Component
+public class DubboGetEnabledRouterMetadata {
```
Comment (CrazyHZM) [inline dubbo-spring-boot/dubbo-spring-boot-compatible/actuator/src/main/java/org/apache/dubbo/spring/boot/actuate/endpoint/metadata/DubboGetEnabledRouterMetadata.java:34]: Named for ServiceOfEnabledRouterSnapshot better, here don't have to add metadata suffix.
The same as below.

**Review by CrazyHZM — CHANGES_REQUESTED** [review]

File: dubbo-spring-boot/dubbo-spring-boot-compatible/actuator/src/main/java/org/apache/dubbo/spring/boot/actuate/endpoint/metadata/DubboOfflineMetadata.java [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+ *
+ * @since 3.3.0
+ */
+@Component
+public class DubboOfflineMetadata extends AbstractDubboMetadata {
+
+    @Autowired
+    public ApplicationModel applicationModel;
+
+    public Map<String, Object> baseOffline(
```
Comment (CrazyHZM) [inline dubbo-spring-boot/dubbo-spring-boot-compatible/actuator/src/main/java/org/apache/dubbo/spring/boot/actuate/endpoint/metadata/DubboOfflineMetadata.java:52]: The operation here should depend on QoS and should not be rewritten.

**Review by CrazyHZM — CHANGES_REQUESTED** [review]

**Pull request closed without merging**